File: pos_next/public/js/purchase.js
frappe.ui.form.on("Purchase Invoice Item", {
    item_code(frm, cdt, cdn) {
        const row = locals[cdt][cdn];

        if (!row.item_code) {
            frappe.model.set_value(
                cdt,
                cdn,
                "custom_selling_amount",
                0
            );
            return;
        }

        frappe.call({
            method: "pos_next.api.purchase.get_standard_selling_rate",
            args: {
                item_code: row.item_code
            },
            callback(r) {
                if (r.message !== undefined) {
                    console.log("Selling amount ----",r)
                    frappe.model.set_value(
                        cdt,
                        cdn,
                        "custom_selling_amount",
                        r.message || 0
                    );
                }
            }
        });
    }
});

File: pos_next/api/purchase.py
import frappe


@frappe.whitelist()
def get_standard_selling_rate(item_code):
    rate = frappe.db.get_value(
        "Item Price",
        {
            "item_code": item_code,
            "price_list": "Standard Selling",
            "selling": 1
        },
        "price_list_rate"
    )

    return rate or 0